modules/ROOT/pages/dataweave-versioning-behavior.adoc

```diff
@@ -0,0 +1,17 @@
+= Versioning Behavior in DataWeave
+
+In each version of DataWeave, changes are introduced that modify the way the language behaves. When existing scripts depend on previous behavior, system properties are introduced to toggle the change so that backwards compatibilty is not broken in the current version. These system properties that control behavior in DataWeave are called *Compatibility Flags*. Each flag defined a deletion version, after which it will no longer exists, and the behavior that it controlled will default to the new one.
```

Comment:
```suggestion
New minor versions of DataWeave can introduce changes that modify the way the language behaves. Starting in DataWeave 2.5, you can use DataWeave system properties (compatibility flags) to continue using existing scripts that depend on the previous behavior. However, each flag defines a deletion version, which identifies the version in which the flag is no longer available and the behavior that it controls defaults to the changed behavior.
```

Reply:
The only thing that I don't know if it's clear from this paragraph is that all compatibility flags are system properties but not all system properties are compatibility flags. We still have properties for different memory/cpu settings that don't have deletion version.

Reply:
@andres-rad Sounds like we need to identify which are which, probably with a deletion version: https://docs.mulesoft.com/dataweave/2.4/dataweave-system-properties 

We also say that some experimental ones are subject to deletion at a future date. 

We should probably something like this to the description and point to the new version doc for more info about how to retain the old version's behavior. 

* Removal version: 2.4

Reply:
I think that this change is good, you can merge if you think the newly autogenerated system properties are ok.

Reply:
@andres-rad Thx, can you commit it so that I can avoid an SoD violation.

Reply:
I will check the autogenerated properties separately. I have a ToDo for that.

Reply:
To self: https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07EE00001IFA61YAH/view for checking this info in autogenerated content.